### PR TITLE
Hint to the runtime to GC after an rAF

### PR DESCRIPTION
**Pull request by asajeffrey**
Hints to the runtime that after an rAF is a good time to GC.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes do not require tests because it's a perf thing

### CODE REVIEW

File: components/script/dom/xrsession.rs [hunk 1]
```diff
@@ -423,6 +428,28 @@ impl XRSession {
...
+    fn gc_hint(&self) {
+        let cx = *self.global().get_cx();
+        let rt = unsafe { JS_GetRuntime(cx) };
+
+        if unsafe { IsIdleGCTaskNeeded(rt) } {
+            #[cfg(feature = "xr-profile")]
+            let gc_start = time::precise_time_ns();
+
+            debug!("GC at the end of an XR rAF");
+            unsafe { JS_GC(cx, GCReason::API) };
```
Comment (jdm) [inline components/script/dom/xrsession.rs:445]: Should we use RunIdleGCTask instead? https://searchfox.org/mozilla-central/source/js/src/jsapi.cpp#1311

Comment (asajeffrey) [reply components/script/dom/xrsession.rs:445]: Ah, that function does almost exactly what we want. It only triggers a minor gc though, but I think that's fine.

Comment (asajeffrey) [reply components/script/dom/xrsession.rs]: Done.

**Review by Manishearth — APPROVED** [review]

**Comment by jdm** [comment]
I would prefer not to merge this until someone has evaluated the performance on a device.

**Comment by asajeffrey** [comment]
Fair enough.

**Comment by asajeffrey** [comment]
I filed a relevant spec issue: https://github.com/immersive-web/webxr/issues/1010

**Pull request closed without merging**